### PR TITLE
Chore: Migrate to use buf for protobuf generation

### PR DESCRIPTION
**Pull request by marefr**
Migrate to use Buf CLI for generating protobuf files. Get rid of custom bash scripts and protoc dependency. 
Also updates plugin SDK to [v0.229.0](https://github.com/grafana/grafana-plugin-sdk-go/releases/tag/v0.229.0).

Noticed we already use buf in enterprise.

Related to https://github.com/grafana/grafana-plugin-sdk-go/pull/979.

### CODE REVIEW

File: pkg/plugins/backendplugin/secretsmanagerplugin/buf.gen.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: v1
+plugins:
+  - plugin: go
+    out: pkg/plugins/backendplugin/secretsmanagerplugin
+    opt: paths=source_relative
+  - plugin: go-grpc
+    out: pkg/plugins/backendplugin/secretsmanagerplugin
+    opt:
+      - paths=source_relative
+      - require_unimplemented_servers=false
```
Comment (DanCech) [inline pkg/plugins/backendplugin/secretsmanagerplugin/buf.gen.yaml:10]: Looks like the existing grpc is generated with this enabled?
```suggestion
      - require_unimplemented_servers=true
```

File: pkg/plugins/backendplugin/pluginextensionv2/buf.gen.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+version: v1
+plugins:
+  - plugin: go
+    out: pkg/plugins/backendplugin/pluginextensionv2
+  - plugin: go-grpc
+    out: pkg/plugins/backendplugin/pluginextensionv2
+    opt:
+      - require_unimplemented_servers=false
```
Comment (DanCech) [inline pkg/plugins/backendplugin/pluginextensionv2/buf.gen.yaml:8]:
```suggestion
      - require_unimplemented_servers=true
```

Comment (DanCech) [reply pkg/plugins/backendplugin/pluginextensionv2/buf.gen.yaml:8]: nit: it's generally preferred to include the trailing newline at the end of the file

Comment (marefr) [reply pkg/plugins/backendplugin/pluginextensionv2/buf.gen.yaml]: I don't think `require_unimplemented_servers=true` makes any difference, but I added it thanks.

**Comment by DanCech** [comment]
Changes themselves look good, but we'll need to add some docs and/or automation to handle installing buf itself

**Comment by marefr** [comment]
@DanCech thanks. https://buf.build/docs/installation don't recommend installing from source so I opted for tooling and docs

File: contribute/backend/upgrading-dependencies.md [hunk 3]
```diff
@@ -2,12 +2,16 @@
...
 
-# Protobuf
+## Protobuf
 
 When upgrading the [protobuf](http://github.com/golang/protobuf) library in Grafana and the plugin SDK,
-you typically also want to upgrade your protobuf compiler toolchain and re-compile protobuf files:
+you typically also want to upgrade your protobuf compiler toolchain and re-compile protobuf files.
 
-```
+**Note:** Uou need Buf CLI installed and availabile in your path, see https://buf.build/docs/installation for instructions.
```
Comment (xnyo) [inline contribute/backend/upgrading-dependencies.md:10]: Typo:
```suggestion
**Note:** You need Buf CLI installed and availabile in your path, see https://buf.build/docs/installation for instructions.
```

**Review by xnyo — APPROVED** [review]
LGTM!

**Comment by marefr** [comment]
✅ https://github.com/grafana/test-datasource with updated SDK (https://github.com/grafana/grafana-plugin-sdk-go/pull/979) on grafana main works as expected
✅ https://github.com/grafana/test-datasource without updated SDK ( v0.213.0) on this branch works as expected

**Review by DanCech — APPROVED** [review]

**Pull request merged**